### DISCUSSION TITLE
CloudFlare Pages Deployment Issue

### DISCUSSION THREAD

**Discussion by ZYGADOX** [vendurehq/vendure#2065] [Q&A]
I am attempting to get a production deployment running on CloudFlare Pages and I have successfully got a deployment working without error but when I try and hit the URL I get a 404 error.

URLs
https://total-furnishing-care-ecom.pages.dev
https://total-furnishing-care-ecom.pages.dev/admin

My build config is…
**Build command:** yarn build
**Build output directory:** /dist
**Root directory:** /
**Build comments on pull requests:** Enabled

I had a lot of trouble getting Pages to run the correct version of node, but adding a **.node-version** file to the root directory with the contents of **16** got me my first successful deployment.

I also added some CloudFlare config to the package.json, but I may have done this wrong; it looks like this…
  "cloudflarePages": {
    "routes": [
      {
        "pattern": "/admin/*",
        "response": {
          "file": "dist/index.html"
        }
      }
    ]
  }

**Comment by michaelbromley** [comment]
Hi,
I don't have a ready answer for this, but the Admin UI is a regular Angular application, and should work the same as any other Angular app on CF Pages - with the caveat that the base url of `/admin` might need some special handling. I've not deployed it to CF pages before so I'm afraid I can't offer any more specific hints right now.

**Reply by ZYGADOX** [reply]
Thanks, Michael. I have tried to deploy to Northflank and DigitalOcean all without much luck (probably my lack of experience). Can you recommend a platform to deploy too?

PS: I absolutely love what you have created and how easy it was to get Vendure up and running locally. I have tried Medusa and Saleor and I think you have outperformed them both and definitely made it a much easier more enjoyable process. Really good work, you and your team should be super happy with your efforts.

Dan

**Reply by michaelbromley** [reply]
Thanks for the kind feedback, I'm really happy to hear that the local dev part at least was smooth - we put a lot of effort into that!

I personally deploy to Northflank - did you try following [our guide](https://www.vendure.io/docs/deployment/deploy-to-northflank/)? That's really the easiest option I know of right now, so I'd be interested to hear what didn't work out for you.

**Comment by ssirec** [comment]
Newbie here.
CloudFlare Pages Issue:

`<html><body>
<pre class="c_kw c_bv c_ack c_xi c_ku c_kv c_df c_aeq c_aer c_aes c_aet c_fu c_fv c_ael c_bx c_aeu">

09:40:58.746 | Warning: /opt/buildhome/repo/packages/admin-ui/src/lib/core/src/shared/components/chart/chart.component.scss exceeded maximum budget. Budget 6.00 kB was not met by 5.40 kB with a total of 11.40 kB.
-- | --
09:40:58.747 |  
09:40:58.747 | Warning: /opt/buildhome/repo/packages/admin-ui/src/lib/core/src/shared/components/rich-text-editor/rich-text-editor.component.scss exceeded maximum budget. Budget 6.00 kB was not met by 2.51 kB with a total of 8.51 kB.

</pre>
</body>
</html>`

followed this Deployment to Northflank https://docs.vendure.io/deployment/deploy-to-northflank

payment method issue:

{
  "kind": "Project",
  "spec": {
    "name": "Vendure Template",
    "region": "us-central",
    "description": "Vendure is a modern, open-source headless commerce framework built with TypeScript & Node.js.",
    "color": "#57637A"
  },
  "response": {
    "status": "failure",
    "error": {
      "code": 409,
      "message": "Please set your default payment method."
    },
    "retries": {
      "attempts": 1,
      "maxAttempts": 1,
      "timestamp": 1690359400
    }
  }
}
![slika](https://github.com/vendure-ecommerce/vendure/assets/125913769/afb898ae-4b6c-4200-995b-025a2146f082)
